packages/codegen-ui-react/lib/utils/forms/storage-field-component.ts

```diff
@@ -0,0 +1,202 @@
+                factory.createIdentifier('imgKeys'),
+                factory.createIdentifier('map'),
+              ),
+              undefined,
+              [
+                factory.createParenthesizedExpression(
+                  factory.createArrowFunction(
+                    undefined,
+                    undefined,
+                    [
+                      factory.createParameterDeclaration(
+                        undefined,
+                        undefined,
+                        undefined,
+                        factory.createIdentifier('s3Key'),
+                        undefined,
+                        undefined,
+                        undefined,
+                      ),
+                    ],
+                    undefined,
+                    factory.createToken(SyntaxKind.EqualsGreaterThanToken),
+                    factory.createParenthesizedExpression(
+                      factory.createObjectLiteralExpression(
+                        [factory.createShorthandPropertyAssignment(factory.createIdentifier('s3Key'), undefined)],
+                        false,
+                      ),
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      );
+    }
+
+    storageManagerAttributes.push(buildOnChangeStatement(component, fieldConfigs));
+    fieldAttributes.push(
+      factory.createJsxAttribute(
+        factory.createIdentifier('errorMessage'),
+        factory.createJsxExpression(
+          undefined,
+          factory.createPropertyAccessChain(
+            errorKey,
+            factory.createToken(SyntaxKind.QuestionDotToken),
+            factory.createIdentifier('errorMessage'),
+          ),
+        ),
+      ),
+      factory.createJsxAttribute(
+        factory.createIdentifier('hasError'),
+        factory.createJsxExpression(
+          undefined,
+          factory.createPropertyAccessChain(
+            errorKey,
+            factory.createToken(SyntaxKind.QuestionDotToken),
+            factory.createIdentifier('hasError'),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Object.entries(component.properties).forEach(([key, value]) => {
+    if (isFieldKey(key)) {
+      if (
+        (key === 'label' && labelDecorator && labelDecorator === 'required' && isRequired) ||
+        (labelDecorator === 'optional' && !isRequired && 'value' in value)
+      ) {
+        fieldAttributes.push(getDecoratedLabel('label', fieldLabel, labelDecorator));
+      } else {
+        fieldAttributes.push(
+          factory.createJsxAttribute(
+            factory.createIdentifier(key),
+            factory.createJsxExpression(undefined, propertyToExpression(componentMetadata, value)),
+          ),
+        );
+      }
+    }
+    if (isStorageManagerKey(key)) {
+      const storageManagerValue =
+        key === 'acceptedFileTypes' ? { ...value, value: (value as any).value.split(',') } : value;
```

Comment:
When the form definition gets passed to the codegen-ui-react, the values get stringified on `mapFormElementProps`. This is a sensitive code that will require more in-depth refactoring to change.

This limits the blast radius to this particular case. We should revisit this later to refactor mapFormElementProps.